src/nodes.coffee

```diff
@@ -497,6 +523,103 @@ exports.Block = class Block extends Base
         fragments.push @makeCode "\n"
     fragments.concat post
 
+  compileComments: (fragments) ->
+    for fragment, fragmentIndex in fragments
+      # Insert comments into the output at the next or previous newline.
+      # If there are no newlines at which to place comments, create them.
+      if fragment.precedingComments
+        # Determine the indentation level of the fragment that we are about
+        # to insert comments before, and use that indentation level for our
+        # inserted comments. Because some fragments begin with a space, e.g.
+        # ` + 1`, we look for either a tab character or at least two spaces
+        # to detect indentation. If someone is actually using one-space
+        # indentation, well, they’ll get unindented line comments.
```

Comment:
It’s looking at the indentation of the previous token’s _generated_ code. In this case, the comment is attached to the `x` fragment, so it’s looking at the previous token’s code, which is `"  return "`—note the two spaces. (For some reason GitHub is showing this to me as one space, but it’s supposed to be `(quote)(space)(space)return(space)(quote)`). That’s how the `// y` line gets two spaces prepended to it.

This is inside `precedingComments`, so then it looks farther back to find the next previous newline, and inserts the `// y` line after it.

Does the CoffeeScript compiler always output JavaScript indented with two spaces? I had assumed it would follow whatever whitespace the original Coffee source had. If it’s always two spaces, that simplifies things considerably.